### ISSUE TITLE
Holiday Snow: make opt-in for visitors

### ISSUE THREAD

**Issue by kraftbj** [Automattic/jetpack#1349]
Quick flyby issue. Add option/default to per-visitor opt-in?

http://www.josheaton.org/responsible-holiday-snow/

Impacted users can shut off snow for any WP.com site they visit when they are logged into their own account, but there is no global option with JP. Perhaps this could be a way to make the feature more accessible?

**Comment by zinigor** [comment]
I'm afraid this one won't make it in time for this year's holiday snow, moving it to Community milestone.

**Comment by johnbillion** [comment]
This is actually a really obnoxious feature on several levels.

A few years ago it may have been cute and entertaining for a small number of people, but I would like to recommend that this functionality is removed completely from Jetpack, and if that's not an option then it should at least carry a warning that its use will piss off most visitors for no good reason.

Reasoning:

 1. CPU usage. This cooks CPUs when using Chrome. Really. It's better in Firefox, but not by a lot.
 2. Accessibility. Moving elements on the page have a great effect on visitors with poor vision.
 3. Apparently, it's even seizure-inducing. [Jetpack admits this itself](https://jeremy.hu/all-about-jetpack-snow/).
 4. If Jetpack has to [defend a feature by explaining how it can be disabled](https://twitter.com/jetpack/status/675298878879584256), maybe it's not such a great feature.

**Comment by jeherve** [comment]
> CPU usage. This cooks CPUs when using Chrome. Really. It's better in Firefox, but not by a lot.

For reference:
https://twitter.com/johnbillion/status/675291197527695361

I couldn't reproduce this. Could you give us more details, so I can try to reproduce? Did you have multiple tabs with snow, by any chance?

> Apparently, it's even seizure-inducing. Jetpack admits this itself.

For reference:
https://aftergadget.wordpress.com/2013/01/23/waspish-wednesday-how-autoplays-let-it-snow-disable-visitors-to-your-website/

> If Jetpack has to defend a feature by explaining how it can be disabled, maybe it's not such a great feature.

You can disable each and every feature in Jetpack, and we're always happy to explain people how to disable a specific feature, whether on Twitter, via email, or in the forums. That doesn't mean we'll remove all features, though. :)

Making that feature opt-in for visitors would be a good first option, though, as we're discussing here. Setting a cookie to allow you to opt out everywhere would be another way to handle this. Happy to hear your opinions on that!

If you think the feature should be completely removed from Jetpack, it's probably best to open a new issue, since this is about changing the default behaviour of the snow.

**Comment by johnbillion** [comment]
> Making that feature opt-in for visitors would be a good first option, though, as we're discussing here.

"I'd love it if it snowed on this web page!" said no-one ever. Know what I mean? :) It's just one of those features that serves no purpose other than to satiate naïve webmasters who thing it's cute and want to add as many features to their website as they can.

> I couldn't reproduce this. Could you give us more details, so I can try to reproduce? Did you have multiple tabs with snow, by any chance?

I'm using Chrome 47 on OS X on a 15 inch MacBook retina. CPU climbs to 99%/100% usage after about a minute of viewing https://norway.wordcamp.org/2016/, which has holiday snow enabled.

> Setting a cookie to allow you to opt out everywhere would be another way to handle this.

That would be handy, but is that an option? Unlike, for example, Gravatar hover cards which can be disabled because it uses a third party script from gravatar.com, holiday snow is hosted locally on each Jetpack powered site which has it enabled. I don't think a global cookie disabling it would be possible.

**Comment by jeherve** [comment]
> no-one ever
> [...]
> naïve webmasters who think it's cute

One could argue that's not no-one. :) I don't use the option on my own site, you don't either, but that doesn't mean no-one wants the option. [There are dozens of Snow plugins in the repo](https://make.wordpress.org/plugins/2015/11/11/tis-the-season-for-snow/), I assume they're there for a reason.

Since the option is enabled on https://norway.wordcamp.org/2016/ , that's at least one person (one of the admins) who wants snow to appear on that site.

> I'm using Chrome 47 on OS X on a 15 inch MacBook retina. CPU climbs to 99%/100% usage after about a minute of viewing https://norway.wordcamp.org/2016/, which has holiday snow enabled.

Thanks! I can't reproduce so far, but I'll keep the tab opened in the background this afternoon, and see if CPU usage starts climbing after a while.

**Comment by johnbillion** [comment]
> There are dozens of Snow plugins in the repo

There are dozens of plugins in the repo that implement full-screen takeover ads and social media popups that cover the content you're trying to read. There's no correlation between quantity (of plugins or downloads) and quality.

> Since the option is enabled on https://norway.wordcamp.org/2016/ , that's at least one person (one of the admins) who wants snow to appear on that site.

I was talking about visitors to the site, not the admins. Think of the following user scenario that would never exist: "As a user, my experience on this website would be improved if it was snowing over the text that I'm trying to read". There's probably even a correlation to "As a driver, my experience driving along this road would be improved if it was snowing and reducing my visibility".

Anyway, I've made my point.

**Comment by johnbillion** [comment]
Looks like it'll be snowing again this year.

**Comment by gravityrail** [comment]
Closed because feature is removed in Jetpack 6.0.

**Comment by jeherve** [comment]
Reopening as snow is back: #9110

**Comment by zinigor** [comment]
#9110 is not going to be merged, closing.

**Comment by zinigor** [comment]
#9110 is not going to be merged, closing.
